source/common/http/filter_manager.cc

```diff
@@ -1466,6 +1511,14 @@ void FilterManager::encodeData(ActiveStreamEncoderFilter* filter, Buffer::Instan
 
     (*entry)->end_stream_ = end_stream && !filter_manager_callbacks_.responseTrailers();
     FilterDataStatus status = (*entry)->handle_->encodeData(data, (*entry)->end_stream_);
+    std::size_t current_filter_idx = std::distance(encoder_filters_.begin(), entry);
+    state_.filters_observed_encode_end_stream_.set(size_t(current_filter_idx),
```

Comment:
do we want to debug assert or envoy bug that that this is only set once to true?

here and elsewhere